src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -248,6 +249,62 @@ public static Path of(URI uri) {
      */
     Path getFileName();
 
+    /**
+     * Returns the file name extension of this path as a {@code String}. The
+     * extension is the portion of the {@linkplain #getFileName file name}
+     * string after the last period character ('.', U+002E FULL STOP). Leading
+     * period characters in the string are ignored. If there is no  extension,
+     * then {@code null} is returned. This will occur if the path has zero
+     * elements ({@link #getFileName()} returns {@code null}), the file name
+     * string does not contain a period character, or only the first character
+     * is a period character. If the last character is a period character but
+     * some other character is not, then the extension is the
+     * {@linkplain String#isEmpty() empty} string.
+     *
+     * @implSpec
+     * The default implementation is equivalent for this path to:
+     * <pre>{@code
+     * int lastPeriod = fileName.lastIndexOf('.');
+     * if (lastPeriod <= 0)
+     *     return null;
+     * return lastPeriod == fileName.length() - 1 ?
+     *     "" :
+     *     fileName.substring(lastPeriod + 1);
+     * }</pre>
+     *
+     * @return  the file name extension of this path, which might be the
+     *          empty string, or {@code null} if no extension is found
+     *
+     * @since 20
+     */
+    default String getExtension() {
+        Path fileName = getFileName();
+        if (fileName == null)
+            return null;
+
+        String fileNameString = fileName.toString();
+        int length = fileNameString.length();
+
+        // An empty or unity length file name string has a null extension
+        if (length > 1) {
+            int lastPeriodIndex = fileNameString.lastIndexOf('.');
+
+            // Indeterminate if there is no period character or
+            // only the first character is a period character
+            if (lastPeriodIndex > 0) {
+                if (lastPeriodIndex == length - 1) {
+                    // null if all period characters, otherwise empty
+                    return fileNameString.matches("\\.{" + length + "}") ?
```

Comment:
Using a regular expression here is probably okay or the initial PR but I could imagine this show up in profiles and needing to be re-implemented.

Reply:
Yeah I don't think we want to use regex here. I'd suggest something like this:

    int lastDot = fileNameString.lastIndexOf('.');
    for (int j = lastDot - 1; j >= 0; j--) {
        if (fileNameString.charAt(j) != '.') {
            return fileNameString.substring(lastDot + 1);
        }
    }
    return null;

Reply:
No longer relevant.